### ISSUE TITLE
injectElements respects "insertafter" only for first element in sequence

### ISSUE THREAD

**Issue by eyalroz** [thundernest/addon-developer-support#20]
In a XUL overlay, you only specify `insertbefore` or `insertafter` for the first element in a sequence of the elements you wish to add. The rest are added after the one whose position you've "pegged" - they're not just appended to the container.

However, it seems like `injectElements()` treats each element independently of the others before it, and thus only those elements with specific `insertbefore` or `insetafter` get inserted before or after. IIANM.

**Comment by jobisoft** [comment]
I cannot confirm this behaviour. Is it documented somewhere?

This is a XUL overlay file used in TB68 with the standard "legacy" overlay loader:
```
<?xml version="1.0"?>
<!DOCTYPE overlay [
<!ENTITY % abMainOverlayDTD SYSTEM "chrome://sendtocategory/locale/catman.dtd" >
%abMainOverlayDTD;
]>
<overlay 
    id="CatManContactPanelOverlay"
    xmlns="http://www.mozilla.org/keymaster/gatekeeper/there.is.only.xul">
    
    <script type="text/javascript" src="chrome://sendtocategory/content/contactpanel/abContactsPanel_overlay.js"/>
   
    <vbox id="results_box" flex="1">
        <vbox id="categoryfilter-box" insertafter="panel-bar">
            <separator class="thin"/>
            <hbox id="categoryfilter-panel-bar" class="toolbar" align="center" insertafter="panel-bar">
                <menulist id="CatManCategoryFilterList"
                oncommand="jbCatMan.contactPanelCategoryMenuChanged();" flex="1"
                persist="value">
                    <menupopup id="CatManCategoryFilterListPopup" writable="true"/>
                </menulist>
 
            </hbox>
        </vbox>
	<label value="TEST" />
    </vbox>

</overlay>
```

As you can see the TEST label is defined right after the vbox with id "categoryfilter-box" with a insertafter position attribute. So the TEST label should be added right after it, according to your request.

This is how the overlayed file looks like:

![grafik](https://user-images.githubusercontent.com/5830621/89105160-f6074380-d41e-11ea-9030-528018e80db8.png)

The TEST label is added to the end of the container, like `injectElements()` would do it. Can you provide a working example to support your statement? Or a link to some sort of documentation?

**Comment by eyalroz** [comment]
Huh.

Hmm.

So, I just checked both my extensions, and it seems I never tried to insert multiple elements except at the end of a container. so I never questioned that what I described is the standard behavior. It's certainly the behavior that makes sense if you don't want to specify insertafter for a all elements in a sequence... but honestly I don't know what's the official policy.

Anyway, in that case... I have another issue I need to open.
